File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/DayPlanMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class DayPlanMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "dayPlanIdPersonalNameFirstName",
            "dayPlanIdPersonalNameLastName",
            "dayPlanIdYear",
            "dayPlanIdMonth",
            "dayPlanIdDay",
            "personVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "Integer",
            "Integer",
            "Integer",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private DayPlanMvoStateEventId value;

    public DayPlanMvoStateEventIdFlattenedDto()
    {
        this(new DayPlanMvoStateEventId());
    }

    public DayPlanMvoStateEventIdFlattenedDto(DayPlanMvoStateEventId value)
    {
        this.value = value;
    }

    public DayPlanMvoStateEventId toDayPlanMvoStateEventId()
    {
        return this.value;
    }


    public String getDayPlanIdPersonalNameFirstName()
    {
        return this.value.getDayPlanId().getPersonalName().getFirstName();
    }

    public void setDayPlanIdPersonalNameFirstName(String dayPlanIdPersonalNameFirstName)
    {
        this.value.getDayPlanId().getPersonalName().setFirstName(dayPlanIdPersonalNameFirstName);
    }

    public String getDayPlanIdPersonalNameLastName()
    {
        return this.value.getDayPlanId().getPersonalName().getLastName();
    }

    public void setDayPlanIdPersonalNameLastName(String dayPlanIdPersonalNameLastName)
    {
        this.value.getDayPlanId().getPersonalName().setLastName(dayPlanIdPersonalNameLastName);
    }

    public Integer getDayPlanIdYear()
    {
        return this.value.getDayPlanId().getYear();
    }

    public void setDayPlanIdYear(Integer dayPlanIdYear)
    {
        this.value.getDayPlanId().setYear(dayPlanIdYear);
    }

    public Integer getDayPlanIdMonth()
    {
        return this.value.getDayPlanId().getMonth();
    }

    public void setDayPlanIdMonth(Integer dayPlanIdMonth)
    {
        this.value.getDayPlanId().setMonth(dayPlanIdMonth);
    }

    public Integer getDayPlanIdDay()
    {
        return this.value.getDayPlanId().getDay();
    }

    public void setDayPlanIdDay(Integer dayPlanIdDay)
    {
        this.value.getDayPlanId().setDay(dayPlanIdDay);
    }

    public Long getPersonVersion()
    {
        return this.value.getPersonVersion();
    }

    public void setPersonVersion(Long personVersion)
    {
        this.value.setPersonVersion(personVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        DayPlanMvoStateEventIdFlattenedDto other = (DayPlanMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Dddml/Wms/Domain/DayPlanMvoStateEventIdFlattenedDto.php
<?php

namespace Dddml\Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class DayPlanMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'dayPlanIdPersonalNameFirstName' => 'string',
            'dayPlanIdPersonalNameLastName' => 'string',
            'dayPlanIdYear' => 'integer',
            'dayPlanIdMonth' => 'integer',
            'dayPlanIdDay' => 'integer',
            'personVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var DayPlanMvoStateEventId
     */
    private $value;

    /**
     * @param DayPlanMvoStateEventId $value
     */
    public function __construct(DayPlanMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new DayPlanMvoStateEventId();
        }
    }

    /**
     * @return DayPlanMvoStateEventId
     */
    public function toDayPlanMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getDayPlanIdPersonalNameFirstName()
    {
        return $this->value->getDayPlanId()->getPersonalName()->getFirstName();
    }

    /**
     * @param string $dayPlanIdPersonalNameFirstName
     */
    public function setDayPlanIdPersonalNameFirstName($dayPlanIdPersonalNameFirstName)
    {
        $this->value->getDayPlanId()->getPersonalName()->setFirstName($dayPlanIdPersonalNameFirstName);
    }

    /**
     * @return string
     */
    public function getDayPlanIdPersonalNameLastName()
    {
        return $this->value->getDayPlanId()->getPersonalName()->getLastName();
    }

    /**
     * @param string $dayPlanIdPersonalNameLastName
     */
    public function setDayPlanIdPersonalNameLastName($dayPlanIdPersonalNameLastName)
    {
        $this->value->getDayPlanId()->getPersonalName()->setLastName($dayPlanIdPersonalNameLastName);
    }

    /**
     * @return integer
     */
    public function getDayPlanIdYear()
    {
        return $this->value->getDayPlanId()->getYear();
    }

    /**
     * @param integer $dayPlanIdYear
     */
    public function setDayPlanIdYear($dayPlanIdYear)
    {
        $this->value->getDayPlanId()->setYear($dayPlanIdYear);
    }

    /**
     * @return integer
     */
    public function getDayPlanIdMonth()
    {
        return $this->value->getDayPlanId()->getMonth();
    }

    /**
     * @param integer $dayPlanIdMonth
     */
    public function setDayPlanIdMonth($dayPlanIdMonth)
    {
        $this->value->getDayPlanId()->setMonth($dayPlanIdMonth);
    }

    /**
     * @return integer
     */
    public function getDayPlanIdDay()
    {
        return $this->value->getDayPlanId()->getDay();
    }

    /**
     * @param integer $dayPlanIdDay
     */
    public function setDayPlanIdDay($dayPlanIdDay)
    {
        $this->value->getDayPlanId()->setDay($dayPlanIdDay);
    }

    /**
     * @return Long
     */
    public function getPersonVersion()
    {
        return $this->value->getPersonVersion();
    }

    /**
     * @param Long $personVersion
     */
    public function setPersonVersion($personVersion)
    {
        $this->value->setPersonVersion($personVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getDayPlanIdPersonalNameFirstName(),
            $this->getDayPlanIdPersonalNameLastName(),
            $this->getDayPlanIdYear(),
            $this->getDayPlanIdMonth(),
            $this->getDayPlanIdDay(),
            $this->getPersonVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return DayPlanMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setDayPlanIdPersonalNameFirstName($pValues[0]);
        $this->setDayPlanIdPersonalNameLastName($pValues[1]);
        $this->setDayPlanIdYear($pValues[2]);
        $this->setDayPlanIdMonth($pValues[3]);
        $this->setDayPlanIdDay($pValues[4]);
        $this->setPersonVersion($pValues[5]);
        return $this;
    }

}

